lang/factory.go

```diff
@@ -22,3 +24,19 @@ func GetLanguage(name string) (core.Language, error) {
 
 	return nil, fmt.Errorf("language not found: %s", name)
 }
+
+func ResolveLanguage(filePath string) (core.Language, error) {
```

Comment:
```suggestion
func ResolveLanguageFromPath(filePath string) (core.Language, error) {
```

Since this is an exported function and package global (i.e. not within a struct receiver), I think we should name it as explicitly as possible.